File: src/0892-Surface-Area-of-3D-Shapes/0892.cpp
class Solution {
public:
    int surfaceArea(vector<vector<int>>& grid) {
        int res = 0, r = grid.size(), c = grid[0].size();
        for (int i = 0; i < r; i++) {
            for (int j = 0; j < c; j++) {
                if (grid[i][j]) res += grid[i][j] * 4 + 2;
                if (i) res -= min(grid[i - 1][j], grid[i][j]) * 2;
                if (j) res -= min(grid[i][j - 1], grid[i][j]) * 2;
            }
        }
        return res;
    }
};

File: src/0892-Surface-Area-of-3D-Shapes/0892.js
var surfaceArea = function(grid) {
    let res = 0, r = grid.length, c = grid[0].length;
    for (let i = 0; i < r; i++) {
        for (let j = 0; j < c; j++) {
            if (grid[i][j] > 0) res += grid[i][j] * 4 + 2;
            if (i > 0) res -= Math.min(grid[i - 1][j], grid[i][j]) * 2;
            if (j > 0) res -= Math.min(grid[i][j - 1], grid[i][j]) * 2;
        }
    }
    return res;
};